### DISCUSSION TITLE
Extending spectral to MWIR (3-5 μm)?

### DISCUSSION THREAD

**Discussion by JackSPearce** [mitsuba-renderer/mitsuba3#471] [Q&A]
Hello! I'm an undergraduate working on an individual research project on rendering/simulating light in the MWIR spectrum. 

I was wondering if anyone familiar with Mitsuba's architecture for the spectral mode could highlight the main obstacles in updating/refitting it for that range (~3000-5000 nm). Any insight would be greatly appreciated 😄

**Comment by njroussel** [comment]
Hi @JackSPearce 

I know very little about the MWIR range. So, it is hard for me to understand your requirements and the potential changes that would be needed.
I know there are a few places where the spectral range is hardcoded, which would definitely need to be changed. A part from that, I would say that there aren't too many other changes needed. It also depends on what you're trying to do.

Maybe @mcrescas has a few other insights ? (he might take a few days to answer)

**Comment by wjakob** [comment]
I haven't used Mitsuba for such applications so far. To my knowledge, one important aspect about longer wavelengths (thermal IR) is that essentially all scene surfaces turn into emitters due to their temperature via black-body emission. This effect could be modeled, though it might be tedious and result in low efficiency of the Monte Carlo sampling process.

Another aspect is that temporal effects play a role: temperature can change over time (via the heat equation). This isn't something that Mitsuba will simulate for you.

You will need to provide tabulated spectral albedos for all materials. The current heuristics to convert RGB values into spectra won't cut it for this wavelength range.
